pkg/agent/controller/traceflow/packetin_test.go

```diff
@@ -453,16 +473,209 @@ func TestParsePacketIn(t *testing.T) {
 				},
 			},
 		},
+		{
+			name:       "packet at source Node forwarded by acnp egress rule",
+			nodeConfig: &config.NodeConfig{},
+			tfState: &traceflowState{
+				name:     "traceflow-pod-to-pod",
+				tag:      1,
+				isSender: true,
+			},
+			pktIn: &ofctrl.PacketIn{
+				PacketIn: &openflow15.PacketIn{
+					TableId: openflow.EgressRuleTable.GetID(),
+					Match: openflow15.Match{
+						Fields: []openflow15.MatchField{*matchTFEgressConjID},
+					},
+					Data: util.NewBuffer(pktBytesPodToPod),
+				},
+			},
+			expectedCalls: func(npQuerier *queriertest.MockAgentNetworkPolicyInfoQuerier, egressQuerier *queriertest.MockEgressQuerier) {
+				npQuerier.EXPECT().GetNetworkPolicyByRuleFlowID(gomock.Any()).Return(
+					&v1beta2.NetworkPolicyReference{
+						Type: v1beta2.AntreaClusterNetworkPolicy,
+						Name: "acnp-1",
+					},
+				)
+				npQuerier.EXPECT().GetRuleByFlowID(gomock.Any()).Return(
```

Comment:
nit: I think that it is pretty easy to know the expected ID here, given that we craft the register data ourselves?
applies to expectations below as well